pkg/cluster/kubernetes_cluster.go

```diff
@@ -97,14 +96,9 @@ func NewKubernetesClusterConnection(connection connection.Connection,
 	}
 
 	k8sCluster := &KubernetesCluster{
-		connection,
-		config,
-		logger,
 		clientset,
 		clientconfig,
 		dynamicClient,
-		io,
-		localizer,
 		kubeconfig,
```

Comment:
I don't understand why connection etc are being removed in favour of being passed as function arguments?